src/components/accession2/Accession2Edit.tsx

```diff
@@ -0,0 +1,181 @@
+import { Grid } from '@mui/material';
+import { Textfield } from '@terraware/web-components';
+import { useEffect, useState } from 'react';
+import { useHistory, useParams } from 'react-router-dom';
+import { Accession2, getAccession2, updateAccession2 } from 'src/api/accessions2/accession';
+import strings from 'src/strings';
+import useDeviceInfo from 'src/utils/useDeviceInfo';
+import useForm from 'src/utils/useForm';
+import FormBottomBar from '../common/FormBottomBar';
+import { APP_PATHS } from 'src/constants';
+
+export default function Accession2Edit(): JSX.Element {
+  const { isMobile } = useDeviceInfo();
+  const { accessionId } = useParams<{ accessionId: string }>();
+  const [accession, setAccession] = useState<Accession2>();
+  const [record, setRecord, onChange] = useForm(accession);
+  const history = useHistory();
+
+  useEffect(() => {
+    const populateAccession = async () => {
+      const response = await getAccession2(parseInt(accessionId, 10));
+      setAccession(response);
+      setRecord(response);
+    };
+
+    if (accessionId !== undefined) {
+      populateAccession();
+    } else {
+      setAccession(undefined);
+    }
+  }, [accessionId, setRecord]);
+
+  const mainStructureSize = isMobile ? 12 : 6;
+
+  const goToAccession2View = () => {
+    if (accession) {
+      history.push(APP_PATHS.ACCESSIONS2_ITEM.replace(':accessionId', accession.id.toString()));
+    }
+  };
+
+  const saveAccession = async () => {
+    if (record) {
+      const response = await updateAccession2(record);
+      if (response.requestSucceeded && accession) {
+        history.push(APP_PATHS.ACCESSIONS2_ITEM.replace(':accessionId', accession.id.toString()));
+      }
+    }
+  };
+
+  return (
+    <Grid container>
+      <Grid container item xs={12} padding={4} paddingBottom={'92px'}>
+        <Grid item xs={12}>
+          <h1>{strings.EDIT_ACCESSION}</h1>
+        </Grid>
+        {accession ? (
+          <Grid container xs={12} spacing={2}>
+            <Grid item xs={mainStructureSize}>
+              <Textfield
```

Comment:
accession number should not be editable yes?

Reply:
I will check in the PRD

Reply:
The properties that can be updated (see generated-schema.ts):
```    
UpdateAccessionRequestPayloadV2: {
      bagNumbers?: string[];
      collectedDate?: string;
      collectionSiteCity?: string;
      collectionSiteCoordinates?: components["schemas"]["Geolocation"][];
      collectionSiteCountryCode?: string;
      collectionSiteCountrySubdivision?: string;
      collectionSiteLandowner?: string;
      collectionSiteName?: string;
      collectionSiteNotes?: string;
      collectionSource?: "Wild" | "Reintroduced" | "Cultivated" | "Other";
      collectors?: string[];
      dryingEndDate?: string;
      facilityId?: number;
      founderId?: string;
      initialQuantity?: components["schemas"]["SeedQuantityPayload"];
      notes?: string;
      plantsCollectedFromMax?: number;
      plantsCollectedFromMin?: number;
      processingMethod?: "Count" | "Weight";
      receivedDate?: string;
      speciesId?: number;
      state?:
        | "Awaiting Check-In"
        | "Pending"
        | "Awaiting Processing"
        | "Processing"
        | "Cleaning"
        | "Processed"
        | "Drying"
        | "Dried"
        | "In Storage"
        | "Withdrawn"
        | "Used Up"
        | "Nursery";
      storageLocation?: string;
      subsetCount?: number;
      subsetWeight?: components["schemas"]["SeedQuantityPayload"];
      targetStorageCondition?: "Refrigerator" | "Freezer";
      viabilityTests?: components["schemas"]["ViabilityTestPayload"][];
      withdrawals?: components["schemas"]["WithdrawalPayload"][];
    };
```

Reply:
i think we should only show the properties that are in the details section? accession id is immutable, we should show it as a label (above the tabs, according to design). i have asked in slack for an initial design of edit view for guidance, in case that helps.